pettingzoo/utils/wrappers/pz_to_gymnasium_wrapper.py

```diff
@@ -0,0 +1,78 @@
+from typing import Callable, Dict, Optional, Protocol, TypeVar, Union
+import gymnasium
+from pettingzoo import AECEnv
+
+
+Action = TypeVar("Action")
+Observation = TypeVar("Observation")
+
+ActOther = Callable[[Observation], Action]
+
+
+class PZ2GymnasiumWrapper(gymnasium.Env):
+  """
+  This class wraps a PettingZoo environment (AECEnv) and exposes a simple Gymnasium environment
+  (for single agent).
+  In order to make it work, one needs to provide in the initialization a mechanism to get the actions
+  of all other agents.
+
+  Note that this class is actually a Gymnasium environment.
+  It is not a PettingZoo wrapper and is neigher a Gymnasium one.
+  """
+
+  def __init__(
+    self,
+    pz_env: AECEnv,
+    act_others: Dict[str, ActOther],
+    take_spaces_from: Union[str, None]
+  ):
+    super().__init__()
+    self._pz_env = pz_env
+    self._act_others = act_others
+    take_spaces_from = (
+      take_spaces_from
+      or next(iter(act_others)) # just use any of the keys in the _act_others dict.
+    )
+    self.observation_space = self._pz_env.observation_space(take_spaces_from)
+    self.action_space = self._pz_env.action_space(take_spaces_from)
+
+  def reset(
+    self,
+    seed: Optional[int] = None,
+    options: Optional[dict] = None,
+  ):
+    super().reset(seed=seed)
+    self._pz_env.reset(seed=seed)
+    self._loop_others()
+    agent = self._pz_env.agent_selection
+    observation, info = (
```

Comment:
I believe you want to call env.last to directly get the observation and info (and reward, termination, truncation) but that's more of a code style thing